src/test/mir-opt/extern-static-ref.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(never_type)]
+#![feature(raw_ref_macros)]
+#![feature(raw_ref_op)]
+
+extern "C" {
+    static FOO: !;
+}
+
+mod foo {
+    #[no_mangle]
+    static FOO: u32 = 5;
+}
+
+#[inline(never)]
+fn bar<T>(_: T) {}
+
+// EMIT_MIR extern_static_ref.main.PreCodegen.after.mir
+fn main() {
+    bar(unsafe { core::ptr::raw_const!(FOO) });
+    bar(unsafe { &raw const FOO });
+}
```

Comment:
Could you also somewhere (hopefully there is a good existing test file?) add a test for the thread-local case? After all this PR also changes thread-local static ptrs quite a bit, would be good to have that covered as well.